packages/lesswrong/lib/routeUtil.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useContext } from 'react';
+import qs from 'qs';
+import { NavigationContext, LocationContext, SubscribeLocationContext } from 'meteor/vulcan:core';
+
+// Given the props of a component which has withRouter, return the parsed query
+// from the URL.
+export function parseQuery(location) {
+  let query = location?.search;
+  if (!query) return {};
+  
+  // The unparsed query string looks like ?foo=bar&numericOption=5&flag but the
+  // 'qs' parser wants it without the leading question mark, so strip the
+  // question mark.
+  if (query.startsWith('?'))
+    query = query.substr(1);
+    
+  return qs.parse(query);
+}
+
+// React Hook which returns the page location (parsed URL and route).
+// Return value contains:
+// {
+//   currentRoute
+//     The object that was passed to addRoute.
```

Comment:
Is this right? It's the string right? I can check this, but only after fixing a bunch of merge conflicts and I'll probably forget.

Reply:
Ah, nope, I'm just mistaken. See the currentRoute.name in the place that I thought was previously just currentRoute.